### PR TITLE
Enhancement/Elemental Shaman - Elemental Blast talent

### PR DESCRIPTION
**Pull request by danny-janse**
The module for Elemental Blast was outdated and over complicated. These changes show the correct total uptime of the elemental blast buffs.

Elemental Blast also has a buff that increases the Haste by 3% this was added to the Haste module. In addition is structured the Haste module into class sections.

### CODE REVIEW

**Review by emallson — APPROVED** [review]
this will likely need a merge conflict resolution in the haste file because of other merged PRs

**Comment by danny-janse** [comment]
I was quite sure it would too. Merge conflict has been resolved.

File: analysis/shaman/src/ElementalBlast.tsx [hunk 1]
```diff
@@ -60,35 +38,38 @@ class ElementalBlast extends Analyzer {
     );
   }
 
-  get elementalBlastUptime() {
-    return this.resultDuration / this.owner.fightDuration;
+  get totalUptime() {
+    return this.critUptime + this.hasteUptime + this.masteryUptime;
```
Comment (HawkCorrigan) [inline analysis/shaman/src/ElementalBlast.tsx:42]: this doesnt work

Comment (HawkCorrigan) [reply analysis/shaman/src/ElementalBlast.tsx:42]: you can have/will have multiple buffs at the same time so you cant just add the uptimes

**Review by HawkCorrigan — CHANGES_REQUESTED** [review]

**Comment by danny-janse** [comment]
I'll close the PR since the fix is just a total buff uptime miscalculation shown in the statistic view. It however causes >100% uptime for elemental shaman.

**Pull request closed without merging**